### ISSUE TITLE
The state of DynamoDB support

### ISSUE THREAD

**Issue by dlras2** [holidayextras/jsonapi-server#211]
Sorry if this isn't appropriate for an issue, but I figured it may be useful to others using this library.

What is the current state of DynamoDB support? The initial branch is nearly a year old. We're happy to work on this ourselves if necessary, but if there was specific problems you ran across when trying to make it work, that would be great info. Any advice on tackling this before we dive in?

**Comment by pmcnr-hx** [comment]
Since this initial work we have implemented data handlers as standalone projects: [`relationaldb`](https://github.com/holidayextras/jsonapi-store-relationaldb) (the most mature of the handlers and the one we're actively using), [`mongodb`](https://github.com/holidayextras/jsonapi-store-mongodb) (quite mature) and [`elasticsearch`](https://github.com/holidayextras/jsonapi-store-elasticsearch). I believe the DynamoDB handler never progressed much further than the experimental stage (@theninj4 can provide further clarification). We could perhaps cherry pick that code into a standalone project and you could then take it from there? We're happy to accept external contributions.

**Comment by dlras2** [comment]
Hi pmcnr-hx,
Pulling out the dynamodb branch into a separate project we could fork and PR would be great. We are currently using jsonapi pretty lightly in production, so we will probably start very thin (simple read and write support), but having a project to work off of would be great!

**Comment by theninj4** [comment]
The current state of the DynamoDB branch is... hmm. I took the empty jsonapi-server handler interface and started pushing around some generic dynamodb code to flesh out what it might eventually look like. I was going in somewhat blind to how DynamoDB should be used - the idea was to get some static code working, then alter it to be dynamic and conform to the resource specification. I shelved the idea when I found the limitation of 5x secondary indexes and ran out of enthusiasm for their vague error messages which really slowed progress. 

TLDR; DynamoDB is great for specifically tailored use cases - making a generic DynamoDB handler for resources of any shape/size is going to be a LOT of hassle.

**Comment by dlras2** [comment]
That is good info to have! Personally I'm new to DynamoDB too, and have only worked with the MongoDB handler. I took a look at the branch and kind of got that impression. Perhaps we will settle on the tailored DynamoDB handler for our use cases, and if we need a more generalized handler we can reassess our options then. Thanks for the reply!
